src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,45 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            BulldozerProducer bulldozer = new BulldozerProducer();
+            return (List<T>) bulldozer.get();
+        } else if (type == Excavator.class) {
+            ExcavatorProducer excavator = new ExcavatorProducer();
+            return (List<T>) excavator.get();
```

Comment:
Casting the result of excavator.get() to List<T> is unsafe and could lead to a ClassCastException at runtime. Instead, use a local variable with a bounded wildcard type, and return a new list created from that variable to ensure type safety.